### PR TITLE
Sp 522 use oa sys auth header

### PR DESCRIPTION
**Pull request by Coupey**

### CODE REVIEW

File: common/middleware/storeOasysSessionKey.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// adds OASys session key to session. This should then be passed as header X-Session-Id in each API call
+
+module.exports = (req, res, next) => {
+  const { sessionKey } = req.query
+  if (sessionKey) {
+    req.session.sessionKey = sessionKey
```
Comment (jonhindle) [inline common/middleware/storeOasysSessionKey.js]: Code style thing. Could just be a oneliner? Your call?
`if (sessionKey) req.session.sessionKey = sessionKey`

Comment (jonhindle) [reply common/middleware/storeOasysSessionKey.js]: Although..... is its omission an error scenario? Shouldn't it always be there?

File: common/middleware/storeOasysSessionKey.test.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  let req
+  let res
+  beforeEach(() => {
+    req = {
+      query: {
+        sessionKey: '',
+      },
+      session: {
+        sessionKey: '',
+      },
```
Comment (jonhindle) [inline common/middleware/storeOasysSessionKey.test.js]: This isn't necessary and not a true reflection of a real request. The lines below should add it in if required 👍

File: common/data/sentencePlanningApi.test.js [hunk 3]
```diff
@@ -43,20 +43,20 @@ describe('sentencePlanningApi', () => {
     nock.cleanAll()
   })
   let mockedEndpoint
-  const token = '1234'
+  const tokens = { authorisationToken: 'mytoken', sessionKey: 'mysessionkey' }
```
Comment (jonhindle) [inline common/data/sentencePlanningApi.test.js]: I think there is a case to be made for having some test constants. That `mockPromise` function is another good candidate. Shall I raise a tech debt ticket?

File: common/middleware/storeOasysSessionKey.js (hunk 1 above)
Comment (Coupey) [reply common/middleware/storeOasysSessionKey.js]: I'm not a fan of one liners (no prob if other people use them though) - I like the safety of the braces! 

Discussed with Stuart when I started this story - we're not assuming there will always be a sessionKey on the query string so no error necessary, just store/update it when it is there and ignore when it isn't.

File: common/data/sentencePlanningApi.test.js (hunk 3 above)
Comment (Coupey) [reply common/data/sentencePlanningApi.test.js]: Yes plz - I agree

Comment (jonhindle) [reply common/data/sentencePlanningApi.test.js]: SP-590 F/E Tech debt - Create test constants raised

**Pull request merged**